prov/gni/test/cq.c

```diff
@@ -852,58 +852,43 @@ Test(cq_wait_set, fd, .init = setup, .disabled = true)
 
 TestSuite(cq_notify_flags, .init = cq_notify_setup, .fini = cq_notify_teardown);
 
-void do_cq_notify(uint64_t op_flags)
+void do_cq_notify(uint64_t flags)
 {
 	ssize_t ret;
 	struct fi_cq_msg_entry entry;
-	uint64_t expected_flags = ~((uint64_t) 0);
 
-	ep->op_flags = op_flags;
-
-	ret = _gnix_cq_add_event(cq_priv, ep, NULL, expected_flags, 0, NULL,
+	ret = _gnix_cq_add_event(cq_priv, ep, NULL, flags, 0, NULL,
 				 0, 0, 0);
 	cr_assert(ret == FI_SUCCESS, "failing in _gnix_cq_add_event");
 
 	ret = fi_cq_read(rcq, &entry, 1);
 	cr_assert(ret == 1, "failing in fi_cq_read");
 
-	switch(op_flags) {
-		case 0:
-			break;
-		case FI_RMA_EVENT:
-			break;
-		case FI_NOTIFY_FLAGS_ONLY:
-			expected_flags &= (FI_REMOTE_CQ_DATA | FI_MULTI_RECV);
-			break;
-		case (FI_RMA_EVENT | FI_NOTIFY_FLAGS_ONLY):
-			expected_flags &= (FI_REMOTE_READ | FI_REMOTE_WRITE |
-					   FI_RMA | FI_REMOTE_CQ_DATA |
-					   FI_MULTI_RECV);
-			break;
-		default:
-			cr_assert(0, "invalid op flags");
+	switch (flags & FI_RMA_EVENT) {
+	case ~FI_RMA_EVENT:
```

Comment:
This won't work, Evan. ~FI_RMA_EVENT is not 0. Just simplify this to an if/else statement.